### PR TITLE
Aman 120/adv filter

### PR DESCRIPTION
**Pull request by gnh374**
# Description  
make adv filter

# Ticket Issue 
[[Provide link to Jira] ](https://a-man-ppl.atlassian.net/browse/AMAN-120) 

# Changes Made  
make adv filter

# Issue Type
- [ ] 🐛 Bug fix 
- [ ] ⚙️ Refactor
- [X] 💻 New feature 
- [ ] 📄 Documentation update
- [ ] ⚠️ Breaking changes
- [ ] 🔍 Testing

# Added/updated tests?

- [ X] Yes
- [ ] No, and this is why: [please replace this line with details on why tests have not been included]

# Additional Notes  
[Include any extra information or considerations for reviewers, such as impacted areas of the codebase]

# (optional) What photo/gif best describes this PR or how it makes you feel?
[For the sake of our sanity and some fun, let's make code reviews more entertaining! 😆🎉]

### CODE REVIEW

**Comment by raviehasan** [comment]
looks great to me, test casesnya juga udh mencakup corner cases nice

**Comment by muhkemallgp** [comment]
looks good mi, test case cover juga edge cases. Paling penamaannya yang bisa lebih straight forward dari title ini dan juga atributnya yang juga sesuai ditentukan sebelumnya ya mi

**Comment by reikathr** [comment]
looking good nom!! test casenya udh bagus

File: src/main/java/rencanakan/id/talentpool/dto/FilterTalentDTO.java [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 @Builder
+@NoArgsConstructor
+@AllArgsConstructor
 @Getter
 @Setter
 public class FilterTalentDTO {
     String name;
+    List<String> skills;
+    List<String> preferredLocations;
+    List<Double> priceRange;
```
Comment (Scarletra) [inline src/main/java/rencanakan/id/talentpool/dto/FilterTalentDTO.java:16]: Ini expected query nya nnti gimana ya? Dari kode kan itu kayak slicing ya, apakah ada case dimana org pengen cek dr min aja atau max aja kayak 1,5 jt keatas dll nya? Better split jd 2 param aja gasih? Min max gitu CMIIW

**Comment by Abbilville** [comment]
Nice nomm menurut ku udah bagus advance filternya, test casenya udah mencakup semua juga

File: src/main/java/rencanakan/id/talentpool/dto/FilterTalentDTO.java (hunk 1 above)
Comment (gnh374) [reply src/main/java/rencanakan/id/talentpool/dto/FilterTalentDTO.java:16]: iya buat yg price cuma terima 2, jadi kalo ga ada minnya masukin 0 aja dri fe. flexible masukin dari fe aja. cuma terima 2 input angka.
